pkg/commands/tools/domain/suggest.go

```diff
@@ -0,0 +1,130 @@
+package domain
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/fastly/cli/pkg/argparser"
+	fsterr "github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/cli/pkg/global"
+	"github.com/fastly/cli/pkg/text"
+	"github.com/fastly/go-fastly/v10/fastly"
+	"github.com/fastly/go-fastly/v10/fastly/domains/v1/tools/suggest"
+)
+
+// GetDomainSuggestionsCommand calls the Fastly API and results domain search results for a given query.
+type GetDomainSuggestionsCommand struct {
+	argparser.Base
+	argparser.JSONOutput
+	// Required.
+	query []string
+	// Optional.
+	defaults argparser.OptionalString
+	keywords argparser.OptionalString
+	location argparser.OptionalString
+	vendor   argparser.OptionalString
+}
+
+// NewDomainSuggestionsCommand returns a usable DomainSuggestionCommand registered under the parent.
+func NewDomainSuggestionsCommand(parent argparser.Registerer, g *global.Data) *GetDomainSuggestionsCommand {
+	cmd := GetDomainSuggestionsCommand{
+		Base: argparser.Base{
+			Globals: g,
+		},
+	}
+
+	cmd.CmdClause = parent.Command("suggest", "Request domain search results for a given query")
+	// Required.
+	cmd.CmdClause.Arg("query", "Search query, e.g. “acme coffee shop”").Required().StringsVar(&cmd.query)
+	// Optional.
+	cmd.CmdClause.Flag("defaults", "Comma-separated list of default zones to include in the search results response, e.g. `--defaults=uk,co.uk`").Action(cmd.defaults.Set).StringVar(&cmd.defaults.Value)
+	cmd.RegisterFlagBool(cmd.JSONFlag())
+	cmd.CmdClause.Flag("keywords", "Comma-separated list of keywords for seeding the search results, e.g. `--keywords=dance,party`").Action(cmd.keywords.Set).StringVar(&cmd.keywords.Value)
+	cmd.CmdClause.Flag("location", "Override IP geolocation with a two-character country code, e.g. `--location=in` to include Indian zones in the search results").Action(cmd.location.Set).StringVar(&cmd.location.Value)
+	cmd.CmdClause.Flag("vendor", "The domain name of a specific registrar or vendor, to filter the search results to the list of zones they support").Action(cmd.vendor.Set).StringVar(&cmd.vendor.Value)
+
+	return &cmd
+}
+
+// Exec invokes the application logic for the command.
+func (g *GetDomainSuggestionsCommand) Exec(_ io.Reader, out io.Writer) error {
+	if g.Globals.Verbose() && g.JSONOutput.Enabled {
+		return fsterr.ErrInvalidVerboseJSONCombo
+	}
+
+	fc, ok := g.Globals.APIClient.(*fastly.Client)
```

Comment:
Fairly minor, but I'd move this after all of the input validation/transformation has been completed.

Reply:
Good catch! I've moved this to after the validation/transformation.